backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Window.java

```diff
@@ -60,28 +60,31 @@ public void invoke (long windowHandle, final boolean focused) {
 			postRunnable(new Runnable() {
 				@Override
 				public void run () {
-					if (windowListener != null) {
-						if (focused) {
-							if (config.pauseWhenLostFocus) {
-								synchronized (lifecycleListeners) {
-									for (LifecycleListener lifecycleListener : lifecycleListeners) {
-										lifecycleListener.resume();
-									}
+					if (focused) {
+						if (config.pauseWhenLostFocus) {
+							synchronized (lifecycleListeners) {
+								for (LifecycleListener lifecycleListener : lifecycleListeners) {
+									lifecycleListener.resume();
 								}
 							}
+							listener.resume();
+						}
+						if (windowListener != null) {
 							windowListener.focusGained();
-						} else {
+						}
+					} else {
+						if (windowListener != null) {
 							windowListener.focusLost();
-							if (config.pauseWhenLostFocus) {
-								synchronized (lifecycleListeners) {
-									for (LifecycleListener lifecycleListener : lifecycleListeners) {
-										lifecycleListener.pause();
-									}
+						}
+						if (config.pauseWhenLostFocus) {
+							synchronized (lifecycleListeners) {
+								for (LifecycleListener lifecycleListener : lifecycleListeners) {
+									lifecycleListener.pause();
 								}
-								listener.pause();
 							}
+							listener.pause();
 						}
-						Lwjgl3Window.this.focused = focused;
+						Lwjgl3Window.this.focused = false;
```

Comment:
I think you accidentally moved this line into the else block. It should also stay as `Lwjgl3Window.this.focused = focused;`.

Reply:
You're right but I'm not sure it was correct before either. Before it was only set if `windowListener != null` which doesn't make sense to me either as the `focus` property wouldn't be updated in some cases right?

Reply:
Yes, that seems to be wrong as well.

Reply:
Ok, changed, unintendedly we have fixed fix another bug in master :)